### PR TITLE
added files for new api version 2017-03-31 (#1)

### PR DESCRIPTION
**Pull request by kejiun**
* added files for new api version 2017-03-31
* only releasing new version number, no api/signature changes

### CODE REVIEW

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 1]
```diff
@@ -0,0 +1,679 @@
...
+				"parameters": [
+					{
+						"$ref": "#/parameters/subscriptionIdParameter"
+					},
+					{
+						"$ref": "#/parameters/apiVersionParameter"
+					}
+				],
+				"responses": {
+					"202": {
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Here you are giving the status 202 which is accepted. But what is the final status? 200? If so, please include it

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 2]
```diff
@@ -0,0 +1,679 @@
...
+				}
+			}
+		},
+		"/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/generateRecommendations/{operationId}": {
+			"get": {
+				"tags": [
+					"GenerateRecommendations"
+				],
+				"description": "Retrieves the status of the recommendation computation or generation process. Invoke this API after calling the generation recommendation. The URI of this API is returned in the Location field of the response header.",
+				"operationId": "Recommendations_GetGenerateRecommendationsStatus",
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Are you particular in this name? Can it be names GenerateRecommendationsStatus_Get which is what we usually recommend

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 3]
```diff
@@ -0,0 +1,679 @@
...
+						"required": true,
+						"type": "string",
+						"format": "uuid"
+					},
+					{
+						"$ref": "#/parameters/apiVersionParameter"
+					}
+				],
+				"responses": {
+					"202": {
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Here you mention 202 which is accepted. Now, how will I poll back to get the response? Which URI to hit? Usually it is obtained from the response of the 202. But there is no response here.

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 4]
```diff
@@ -0,0 +1,679 @@
...
+				}
+			}
+		},
+		"/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}": {
+			"get": {
+				"tags": [
+					"Suppressions"
+				],
+				"description": "Obtains the details of a suppression.",
+				"operationId": "Suppressions_Get",
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Does this operation provide a list of suppressions for one recommendations? In that case, the recommendation is to use the id Suppressions_ListByRecommendations

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 5]
```diff
@@ -0,0 +1,679 @@
...
+				"description": "Lists all the available Advisor REST API operations.",
+				"operationId": "Operations_List",
+				"parameters": [
+					{
+						"$ref": "#/parameters/apiVersionParameter"
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK.",
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Provide a valid description

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 6]
```diff
@@ -0,0 +1,679 @@
...
+						"name": "$skipToken",
+						"in": "query",
+						"description": "The page-continuation token to use with a paged version of this API.",
+						"required": false,
+						"type": "string"
+					}
+				],
+				"responses": {
+					"200": {
+						"description": "OK.",
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Provide a vaid description

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 7]
```diff
@@ -0,0 +1,679 @@
...
+				"properties": {
+					"$ref": "#/definitions/RecommendationProperties",
+					"description": "The properties of the recommendation.",
+					"x-ms-client-flatten": true
+				},
+				"suppressionIds": {
+					"description": "The list of snoozed and dismissed rules for the recommendation.",
+					"type": "array",
+					"items": {
+						"format": "uuid",
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Is it possible to avoid usage of uuids?

File: arm-advisor/2017-03-31/swagger/advisor.json [hunk 8]
```diff
@@ -0,0 +1,679 @@
...
+				"value": {
+					"description": "The list of recommendations.",
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/ResourceRecommendationBase"
+					}
+				}
+			}
+		},
+		"ResourceRecommendationBase": {
```
Comment (sarangan12) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Can you confirm if this is a resource? If so, it should have an allOf of a resource.

**Review by sarangan12 — CHANGES_REQUESTED** [review]

**Comment by sarangan12** [comment]
Running the swagger against linter, I get the following errors:

ERROR: Per the Noun_Verb convention for Operation Ids, the noun 'Recommendations' should not appear after the underscore.
        Path: ..\advisor.json#/paths/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/generateRecommendations/{operationId}/get/operationId
ERROR: Top level properties should be one of name, type, id, location, properties, tags, plan, sku, etag, managedBy, identity. Extra properties found: "SuppressionContract/suppressionId, SuppressionContract/ttl".
        Path: ..\advisor.json#/paths/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}
ERROR: Tracked Resource failing validation is: "SuppressionContract". Validation Failed: 2.
    A Tracked Resource must have:
    1. A Get Operation
    2. A ListByResourceGroup operation with x-ms-pageable extension and
    3. A ListBySubscriptionId operation with x-ms-pageable extension.
    4. "type","location","tags" should not be used in the RP property bag named "properties".
        Path: ..\advisor.json#/definitions
ERROR: Tracked resource 'SuppressionContract' must have patch operation that at least supports the update of tags.
        Path: ..\advisor.json#/definitions


Please fix them

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 4 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: this one returns only one suppression

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 7 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: may I have the reason

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 8 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: this is a extension resource, does it count?

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 4 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: this one return single suppressions

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 7 above)
Comment (ravbhatnagar) [inline arm-advisor/2017-03-31/swagger/advisor.json]: Hank and I had a conversation on this. The design of this API will be revisited after GA. There may not be a need for suppressionIds. But lets not block GA due to this. This is read-only for now and service generates it. So user does not have to pass in the GUID.

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 1 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: The design of this API will be revisited after GA.

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 3 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: The design of this API will be revisited after GA.

File: arm-advisor/2017-03-31/swagger/advisor.json (hunk 8 above)
Comment (kejiun) [inline arm-advisor/2017-03-31/swagger/advisor.json]: done

**Comment by sarangan12** [comment]
Based on the offline communications, I see some of the API design issues will be handled after GA release. The documentation issues seem to be resolved now. Approving these

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**